robot state publisher failing on buildfarm

http://www.ros.org/debbuild/hydro.html?q=robot_state_publisher

Extract from the build log:

```
Linking CXX executable devel/lib/robot_state_publisher/robot_state_publisher
/usr/bin/cmake -E cmake_link_script CMakeFiles/robot_state_publisher.dir/link.txt --verbose=1
/usr/lib/ccache/c++   -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Wformat-security     CMakeFiles/robot_state_publisher.dir/src/joint_state_listener.cpp.o  -o devel/lib/robot_state_publisher/robot_state_publisher -rdynamic devel/lib/librobot_state_publisher_solver.so -llog4cxx /opt/ros/hydro/lib/libroscpp.so -lpthread -lboost_signals-mt -lboost_filesystem-mt -lboost_system-mt /opt/ros/hydro/lib/libcpp_common.so /opt/ros/hydro/lib/libroscpp_serialization.so /opt/ros/hydro/lib/librostime.so -lboost_date_time-mt -lboost_thread-mt /opt/ros/hydro/lib/librosconsole.so -lboost_regex-mt -llog4cxx /opt/ros/hydro/lib/libxmlrpcpp.so /opt/ros/hydro/lib/libtf.so /opt/ros/hydro/lib/libmessage_filters.so /opt/ros/hydro/lib/libtf2_ros.so /opt/ros/hydro/lib/libactionlib.so /opt/ros/hydro/lib/libtf2.so /opt/ros/hydro/lib/libconsole_bridge.so /opt/ros/hydro/lib/libtf_conversions.so /opt/ros/hydro/lib/libkdl_conversions.so /opt/ros/hydro/lib/libkdl_parser.so -ltinyxml /opt/ros/hydro/lib/liburdf.so /opt/ros/hydro/lib/liburdfdom_sensor.so /opt/ros/hydro/lib/liburdfdom_model_state.so /opt/ros/hydro/lib/liburdfdom_model.so /opt/ros/hydro/lib/liburdfdom_world.so /opt/ros/hydro/lib/librosconsole_bridge.so -Wl,-rpath,/tmp/buildd/ros-hydro-robot-state-publisher-1.9.9-0precise-20131207-0746/obj-x86_64-linux-gnu/devel/lib:/opt/ros/hydro/lib: 
/usr/bin/ld: CMakeFiles/robot_state_publisher.dir/src/joint_state_listener.cpp.o: undefined reference to symbol 'KDL::Tree::Tree(std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: note: 'KDL::Tree::Tree(std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)' is defined in DSO /opt/ros/hydro/lib/liborocos-kdl.so.1.2 so try adding it to the linker command line
/opt/ros/hydro/lib/liborocos-kdl.so.1.2: could not read symbols: Invalid operation
collect2: ld returned 1 exit status
make[4]: *** [devel/lib/robot_state_publisher/robot_state_publisher] Error 1
```

http://jenkins.ros.org/view/HbinP64/job/ros-hydro-robot-state-publisher_binarydeb_precise_amd64/41/console

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 2.8)
 project(robot_state_publisher)
 
 find_package(Eigen REQUIRED)
-
+find_package(orocos_kdl REQUIRED) 
 find_package(catkin REQUIRED
   COMPONENTS roscpp rosconsole rostime tf tf_conversions kdl_parser
 )
@@ -10,23 +10,24 @@ find_package(catkin REQUIRED
 catkin_package(
   LIBRARIES ${PROJECT_NAME}_solver
   INCLUDE_DIRS include
-  DEPENDS roscpp rosconsole rostime tf tf_conversions kdl_parser
+  DEPENDS roscpp rosconsole rostime tf tf_conversions kdl_parser orocos_kdl
 )
 
 include_directories(SYSTEM ${EIGEN_INCLUDE_DIRS})
-include_directories(include ${catkin_INCLUDE_DIRS})
+include_directories(include ${catkin_INCLUDE_DIRS} ${orocos_kdl_INCLUDE_DIRS})
+link_directories(${orocos_kdl_LIBRARY_DIRS})
 
 add_library(${PROJECT_NAME}_solver
   src/robot_state_publisher.cpp src/treefksolverposfull_recursive.cpp
 )
-target_link_libraries(${PROJECT_NAME}_solver ${catkin_LIBRARIES})
+target_link_libraries(${PROJECT_NAME}_solver ${catkin_LIBRARIES} ${orocos_kdl_LIBRARIES})
 
 add_executable(${PROJECT_NAME} src/joint_state_listener.cpp)
-target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}_solver log4cxx)
+target_link_libraries(${PROJECT_NAME} ${PROJECT_NAME}_solver log4cxx ${orocos_kdl_LIBRARIES})
 
 # compile the same executable using the old name as well
 add_executable(state_publisher src/joint_state_listener.cpp)
-target_link_libraries(state_publisher ${PROJECT_NAME}_solver log4cxx)
+target_link_libraries(state_publisher ${PROJECT_NAME}_solver log4cxx ${orocos_kdl_LIBRARIES})
 
 # CATKIN has no ROS test for now. Disabling
 

--- a/package.xml
+++ b/package.xml
@@ -22,6 +22,7 @@
 
   <build_depend>eigen</build_depend>
   <build_depend>kdl_parser</build_depend> 
+  <build_depend>orocos_kdl</build_depend>
   <build_depend>rosconsole</build_depend>
   <build_depend>roscpp</build_depend>
   <build_depend>rostime</build_depend>
@@ -32,6 +33,7 @@
   <run_depend>catkin</run_depend>
   <run_depend>eigen</run_depend>
   <run_depend>kdl_parser</run_depend> 
+  <run_depend>orocos_kdl</run_depend>
   <run_depend>rosconsole</run_depend>
   <run_depend>roscpp</run_depend>
   <run_depend>rostime</run_depend>

